common/pulp/common/error_codes.py

```diff
@@ -122,6 +122,9 @@
 PLP0042 = Error("PLP0042", _("This request is forbidden."), [])
 PLP0043 = Error("PLP0043", _("Database 'write_concern' config can only be 'majority' or 'all'. "
                              "Refer to /etc/pulp/server.conf for proper use."), [])
+PLP0044 = Error("PLP0044", _("Content unit type '%(content_unit)s' explicitly defines an index for"
+                             "its unit key. This is not allowed because the platform handles it for"
+                             "you."), ['content_unit'])
```

Comment:
@dkliban Incorporate @rbarlow 's feedback how you would like. He makes a good point. It should be noted that there already exist coded exceptions that are develop focused, so we're already in the state he suggest we shouldn't be.